vnext/src/Libraries/Components/Button.windows.js

```diff
@@ -335,7 +335,67 @@ class Button extends React.Component<
     //  Platform.OS === 'android' ? TouchableNativeFeedback : TouchableOpacity;
     // Windows]
     // [Windows
-    if (Platform.OS === 'windows') {
+    if (Platform.OS === 'windows' && color) {
+      return (
+        <Touchable
+          accessibilityLabel={accessibilityLabel}
+          accessibilityRole="button"
+          accessibilityState={accessibilityState}
+          hasTVPreferredFocus={hasTVPreferredFocus}
+          nextFocusDown={nextFocusDown}
+          nextFocusForward={nextFocusForward}
+          nextFocusLeft={nextFocusLeft}
+          nextFocusRight={nextFocusRight}
+          nextFocusUp={nextFocusUp}
+          testID={testID}
+          disabled={disabled}
+          onPress={onPress}
+          tabIndex={tabIndex}
+          touchSoundDisabled={touchSoundDisabled}
+          underlayColor="white"
+          onShowUnderlay={() => {
+            this.setState({pressed: true});
+          }}
+          onHideUnderlay={() => {
+            this.setState({pressed: false});
+          }}
+          onMouseEnter={() => {
+            if (!disabled) this.setState({hover: true});
+          }}
+          onMouseLeave={() => {
+            if (!disabled) this.setState({hover: false});
+          }}>
+          <View
+            style={
+              this.state.pressed
+                ? [
+                    buttonStyles,
+                    {
+                      borderColor: PlatformColor('ButtonBorderBrushPressed'),
+                      borderBottomWidth: 1,
+                    },
+                  ]
+                : buttonStyles
+            }>
+            <View
+              style={
+                this.state.hover
+                  ? [
+                      {
+                        backgroundColor: 'rgba(255, 255, 255, 0.1)',
+                        borderRadius: 3,
```

Comment:
Can we avoid duplicating this between the existing style?

Reply:
The large duplication in code is definitely clunky. Unfortunately the default cause and the "color" button case require adjustments in how they are setup based on coloring limitations. 

For the default button we want to grab the XAML Button colors for text and button background at all states (pressed, rest, hover, disabled). To do that we need to setup the button via TouchableHighlight, Views, and Text such that the XAML colors are shown purely and there is no layering of color. 

For the button with color specified, we are given the ColorValue color and then we need to mock the change in button color for all the states. Since there isn't a way in RN to take a color and build out a new color from that color with changes such as lightening/deepening the color, we have to rely on layering color and TouchableHighlight's activeOpacity functionality to generate slight visual changes in the color. This is how we get line 385 here. To mock the hover state which should appear as a slight lightening of the button, I colored a View with a translucent white which lays on top of the original color when the button is in hover state.